openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/ActivityFeedCardV2/FeedCardFooter/FeedCardFooter.tsx

```diff
@@ -57,11 +64,71 @@ function FeedCardFooter({
 
     return { latestReplyTimeStamp, repliedUniqueUsersList };
   }, [feed?.posts]);
+  const { updateEntityThread } = useActivityFeedProvider();
+  const updateAnnouncementsThreadReactions = async (
+    post: Post,
+    reactionType: ReactionType,
+    reactionOperation: ReactionOperation
+  ) => {
+    let updatedReactions = isPost ? post.reactions || [] : feed.reactions || [];
+    if (reactionOperation === ReactionOperation.ADD) {
+      const reactionObject = {
+        reactionType,
+        user: {
+          id: currentUser?.id as string,
+        },
+      };
+
+      updatedReactions = [...updatedReactions, reactionObject as Reaction];
+    } else {
+      updatedReactions = updatedReactions.filter(
+        (reaction) =>
+          !(
+            reaction.reactionType === reactionType &&
+            reaction.user.id === currentUser?.id
+          )
+      );
+    }
+    const originalObject = isPost
+      ? { ...post, reactions: [...(post.reactions || [])] }
+      : { ...feed, reactions: [...(feed.reactions || [])] };
+
+    const updatedObject = isPost
+      ? { ...post, reactions: updatedReactions }
+      : { ...feed, reactions: updatedReactions };
+
+    const patch = compare(originalObject, updatedObject);
 
+    if (!isEmpty(patch)) {
+      if (feed.type === ThreadType.Announcement && !isAnnouncementTab) {
+        if (isPost) {
```

Comment:
Seems too many conditions can we simplify globally tackling condition

Reply:
done